src/index.js

```diff
@@ -168,6 +179,37 @@ export default {
             });
     },
 
+    _prepareChromeCapabilities (capabilities) {
+        if (process.env['BROWSERSTACK_CHROME_ARGS'] && process.env['BROWSERSTACK_CHROME_ARGS'].length > 0)
+            capabilities.chromeOptions = { args: [process.env['BROWSERSTACK_CHROME_ARGS']] };
+    },
+
+    async _prepareFirefoxCapabilities (capabilities) {
+        if (!process.env['BROWSERSTACK_USE_AUTOMATE'])
+            return;
+
+        const FirefoxProfile = require('firefox-profile');
+        const profile        = new FirefoxProfile();
+
+        profile.defaultPreferences = {};
+
+        profile.setPreference('browser.helperApps.neverAsk.saveToDisk', getMimeTypes());
+        profile.updatePreferences();
+
+        capabilities['firefox_profile'] = await this._encodeFirefoxProfile(profile);
+    },
+
+    async _encodeFirefoxProfile (profile) {
+        return new Promise((resolve, reject) => {
+            profile.encoded(function (err, encodedProfile) {
```

Comment:
I guess we can use the `util.promisify` here.
```js
async _encodeFirefoxProfile (profile) {
   const promisifiedProfileEncoded = util.promisify(profile.encoded);
  
   return promisifiedProfileEncoded(); 
}
```

Or combine call as following:
```js
capabilities['firefox_profile'] = await util.promisify(profile.encoded)();
```